eureka_ml_insights/models/models.py

```diff
@@ -148,31 +189,36 @@ def create_request(self, text_prompt, query_images=None, system_message=None):
         headers = {
             "Content-Type": "application/json",
             "Authorization": ("Bearer " + self.api_key),
-            "azureml-model-deployment": self.model_name,
         }
 
         return urllib.request.Request(self.url, body, headers)
 
     def get_response(self, request):
+        # Get the model response and measure the time taken.
+        start_time = time.time()
         response = urllib.request.urlopen(request)
+        end_time = time.time()
+        # Parse the response and return the model output.
         res = json.loads(response.read())
-        return res["output"]
+        self.model_output = res["output"]
+        self.response_time = end_time - start_time
```

Comment:
Maybe let's declare self.model_output and self.response_time together with the other variables so that the reader knows that these are important class attributes that will be set/used later.

Reply:
If you look at the Model class, they are declared together :)